### DISCUSSION TITLE
Running proj more than once

### DISCUSSION THREAD

**Discussion by JaneSullivan-NOAA** [afsc-assessments/AK-projection-simulation#12] [General]
## Mean_Catch in percentiles.out

For our MESA rockfish "author's F" proj scenario (our second run of proj), we set future catches (Yr+1 and Yr+2) in spp_catch.dat using the Mean_Catch value from the percentiles.out file produced by running the "max F" proj scenario (when we don't set future catches). Can you help me understand what Mean_Catch is? @jimianelli 

![image](https://user-images.githubusercontent.com/68303089/227312387-8829fcb7-6451-4cd9-aebf-2644503f0ba5.png)

**Comment by jimianelli** [comment]
for near term years there's almost no variability due to recruitment
(starts w/ most recent numbers at age estimates).

For each simulation, for years where constant catch isn't specified, catch
will be a function of the alternative and the F and vary due to recruitment
variability and the age structure. the mean_catch column is the mean of
those catches across simulations.

But maybe we shouldn't use this file any longer as mentioned before?

On Thu, Mar 23, 2023 at 3:28 PM Jane Sullivan ***@***.***>
wrote:

> For our MESA rockfish "author's F" proj scenario (our second run of proj),
> we set future catches (Yr+1 and Yr+2) in spp_catch.dat using the Mean_Catch
> value from the percentiles.out file produced by running the "max F" proj
> scenario (when we don't set future catches). Can you help me understand
> what Mean_Catch is? @jimianelli <https://github.com/jimianelli>
>
> [image: image]
> <https://user-images.githubusercontent.com/68303089/227312387-8829fcb7-6451-4cd9-aebf-2644503f0ba5.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/afsc-assessments/spmR/discussions/12>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAUW7YUS7YOUKKIVBWH4WSLW5SI6HANCNFSM6AAAAAAWFSRG6A>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Jim Ianelli

**Reply by JaneSullivan-NOAA** [reply]
Thanks @jimianelli. For now I'm just trying to understand/document what we do now. 
So because Alternative 1 sets F= maxFABC, "Alt 1 Mean_Catch from Max F scenario’s percentiles.out" is the same as saying "max ABC"? If that's the case, what is Cabc in the percentiles.out file?

![image](https://user-images.githubusercontent.com/68303089/227347129-d055d263-7484-4429-8530-898c1364a4cc.png)

**Comment by jimianelli** [comment]
Cabc is the point estimate if the stock is at B40, for Alt 1 (if I remember
correctly). it's not time varying, it's a point estimate

On Thu, Mar 23, 2023 at 5:33 PM Jane Sullivan ***@***.***>
wrote:

> Thanks @jimianelli <https://github.com/jimianelli>. For now I'm just
> trying to understand/document what we do now.
> So because Alternative 1 sets F= maxFABC, "Alt 1 Mean_Catch from Max F
> scenario’s percentiles.out" is the same as saying "max ABC"? If that's the
> case, what is Cabc in the percentiles.out file?
>
> [image: image]
> <https://user-images.githubusercontent.com/68303089/227347129-d055d263-7484-4429-8530-898c1364a4cc.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/afsc-assessments/spmR/discussions/12#discussioncomment-5411091>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAUW7YUZNDOD2MSBVYEFQZTW5SXSPANCNFSM6AAAAAAWFSRG6A>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Jim Ianelli

**Reply by JaneSullivan-NOAA** [reply]
super! thanks.

ok, since we're on a roll... do you know why we run proj twice (max F and author's F as shown in the table above) instead of using the `#_Author_F_as_fraction_F_40%` option in the species-specific data file? it looks like this gets read in as `FABC_Adj` and is used in Scenario 2 (Author's F):
![image](https://user-images.githubusercontent.com/68303089/227390408-bdf07b3c-cbb7-411b-b18d-f2761a677346.png)
![image](https://user-images.githubusercontent.com/68303089/227390524-a346f04b-7d52-49c1-a3e2-fd928aceeac8.png)

**Comment by jimianelli** [comment]
Easy one, adjust is to F, fixed catch is F that gives that catch

On Thu, Mar 23, 2023 at 9:04 PM Jane Sullivan ***@***.***>
wrote:

> super! thanks.
>
> ok, since we're on a roll... do you know why we run proj twice (max F and
> author's F as shown in the table above) instead of using the
> #_Author_F_as_fraction_F_40% option in the species-specific data file? it
> looks like this gets read in as FABC_Adj and is used in Scenario 2
> (Author's F):
> [image: image]
> <https://user-images.githubusercontent.com/68303089/227390408-bdf07b3c-cbb7-411b-b18d-f2761a677346.png>
> [image: image]
> <https://user-images.githubusercontent.com/68303089/227390524-a346f04b-7d52-49c1-a3e2-fd928aceeac8.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/afsc-assessments/spmR/discussions/12#discussioncomment-5412250>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAUW7YWVHWSIBROXJ3DF7KDW5TQHBANCNFSM6AAAAAAWFSRG6A>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
Jim Ianelli

**Reply by JaneSullivan-NOAA** [reply]
But the catch that comes out of the max F scenario is based on F_maxABC, so is there a functional difference? I would think the Year+1 values should be the same but the Year+2 values would be more accurate if we use the Alt 2 built in feature because the F_maxABC for Year+2 would be informed by Year+1.

**Comment by jimianelli** [comment]
¯\_(ツ)_/¯

On Thu, Mar 23, 2023 at 9:19 PM Jane Sullivan ***@***.***>
wrote:

> But the catch that comes out of the max F scenario is based on F_maxABC,
> so is there a functional difference? I would think the Year+1 values should
> be the same but the Year+2 values would be more accurate if we use the Alt
> 2 built in feature because the F_maxABC for Year+2 would be informed by
> Year+1.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/afsc-assessments/spmR/discussions/12#discussioncomment-5412306>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAUW7YTC2MKT4ZWIF5OHUTLW5TSCDANCNFSM6AAAAAAWFSRG6A>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
Jim Ianelli

**Reply by JaneSullivan-NOAA** [reply]
Here's a comparison of the two approaches. I'm inclined to recommend we use the built-in approach instead of the "run twice" approach. You get really similar results, but the built-in approach is more informed (accounts for the cumulative effect of not fishing at F40) and is consistent throughout the full projection time period. Is there something I'm missing here?
![image](https://user-images.githubusercontent.com/68303089/227682502-7d0282fb-2b59-4254-95e0-2acd2017278b.png)

**Comment by jimianelli** [comment]
lost me a bit here, hard to tell w/o seeing what was specified for number
of fixed years of catch (and what those catches were).

And yes, probably polishing peanuts if "assumed catches" are close to
author's F

On Fri, Mar 24, 2023 at 11:40 PM Jane Sullivan ***@***.***>
wrote:

> Here's a comparison of the two approaches. I'm inclined to recommend we
> use the built-in approach instead of the "run twice" approach. You get
> really similar results, but the built-in approach is more informed
> (accounts for the cumulative effect of not fishing at F40) and is
> consistent throughout the full projection time period. Is there something
> I'm missing here?
> [image: image]
> <https://user-images.githubusercontent.com/68303089/227682502-7d0282fb-2b59-4254-95e0-2acd2017278b.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/afsc-assessments/spmR/discussions/12#discussioncomment-5424280>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAUW7YQ4XRK7V3IVF6CFZ5DW5ZLLTANCNFSM6AAAAAAWFSRG6A>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Jim Ianelli

**Reply by JaneSullivan-NOAA** [reply]
Ok so now I think I understand what's going on a little better. In the built-in Alt 2, the ABC is set to the author's F=FABC, so even though the yields are roughly the same under both approaches, the ABCs are different. In the example below, you can see how different the ABCs are but the OFLs are roughly the same (they should be identical in the first projection year, then start to diverge). This seems like something we should change in the code, or at least output NAA so we can do the calculations in R. Open to thoughts on this. It seems worth cataloguing the different assumptions people use... for example, I was a little surprised by how different the EBS Pcod scenarios are to GOA REBS. 
![image](https://user-images.githubusercontent.com/68303089/227741577-ce63ac4e-aadb-4169-b9d8-bef9e23773a4.png)
Also since you asked, the catch assumptions are below (catch = observed, catch_used = in-year extrapolated catch using the endyr_ratio or future catches defined using the yield_rate (TAC/ABC averaged over the last three full yrs):
![image](https://user-images.githubusercontent.com/68303089/227743674-dc3e14cf-0c8d-43f6-8951-eae81f63cde0.png)

**Comment by jimianelli** [comment]
also this dataframe isn't output from the admb code and isn't exactly
correct (in the labels). This is the reason that there's a Catch at maximum
permissible FABC and an assumed catch (in this case for 2021 and 2022). For
such updates it seems best for communication purposes to put NA for any
historic year since the F's for scenarios specifically aren't applied. Just
a thought.

On Fri, Mar 24, 2023 at 11:40 PM Jane Sullivan ***@***.***>
wrote:

> Here's a comparison of the two approaches. I'm inclined to recommend we
> use the built-in approach instead of the "run twice" approach. You get
> really similar results, but the built-in approach is more informed
> (accounts for the cumulative effect of not fishing at F40) and is
> consistent throughout the full projection time period. Is there something
> I'm missing here?
> [image: image]
> <https://user-images.githubusercontent.com/68303089/227682502-7d0282fb-2b59-4254-95e0-2acd2017278b.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/afsc-assessments/spmR/discussions/12#discussioncomment-5424280>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAUW7YQ4XRK7V3IVF6CFZ5DW5ZLLTANCNFSM6AAAAAAWFSRG6A>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Jim Ianelli

**Reply by JaneSullivan-NOAA** [reply]
@jimianelli catch assumptions are above! you're right, this table doesn't come from the ADMB code. Its an extended version of the scenario summary table the GOA rockfish authors put in all our assessments. For the second run of proj, we specify the the last year's catch (only in a partial year), the extrapolated in-year catch, and two future years of catch.

**Reply by JaneSullivan-NOAA** [reply]
Let's discuss this next week... I'm not sure I'm totally following you here: "For such updates it seems best for communication purposes to put NA for any historic year since the F's for scenarios specifically aren't applied."

**Comment by jimianelli** [comment]
well for the years with specified catch, the column titles are incorrect
(and should be NA because the values in there are the assumed catch)

On Sat, Mar 25, 2023 at 6:50 PM Jane Sullivan ***@***.***>
wrote:

> Let's discuss this next week... I'm not sure I'm totally following you
> here: "For such updates it seems best for communication purposes to put NA
> for any historic year since the F's for scenarios specifically aren't
> applied."
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/afsc-assessments/spmR/discussions/12#discussioncomment-5428987>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAUW7YS6MGCDY7I464N2Z2LW55SA3ANCNFSM6AAAAAAWFSRG6A>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Jim Ianelli
